src/index.tsx

```diff
@@ -970,6 +970,41 @@ class App extends React.Component<{}, AppState> {
                 elementAtPosition.y +
                 CANVAS_WINDOW_OFFSET_TOP +
                 elementAtPosition.height / 2;
+            } else if (!e.altKey) {
+              // Element must be searched from the end
+              // to pick element with the highest z-index
+              const elementClickedInside = elements
+                .slice()
+                .reverse()
```

Comment:
I would prefer not to do that, there can be a lot of elements and this is doing two expensive copy of a potentially big array. If you can write a normal loop that would be better.

Reply:
I think considering what we're doing on each render, `slice` & `reverse` (irrespective of array size) is the least of our perf concerns :), but why not.

Reply:
Perf is going to be a concern for this kind of application. We should avoid introducing obviously bad things perf wise when it doesn’t need to.

We’re not that inefficient right now, all the shape draw methods are precomputed. So we’re doing one pass to draw things and one pass for hit testing.

I don’t believe that we allocate that many things on those two passes.

We’ll want to implement some sort of frustrum culling to not draw things out of screen and make hit test faster by using bounds approximations but let’s worry about that for another day.

Reply:
For the draw commands, if we start using webgl, we can send draw commands to the gpu upfront and reuse them, so we don't have to send that entire list all the time. I've seen massive speedups when doing that.